### PR TITLE
aws sdk adapts to Tencent cos

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] bug

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

When using the tencent, aws sdk config need to add field 'region'.

### CODE REVIEW

File: be/src/env/env_s3.cpp [hunk 1]
```diff
@@ -204,6 +204,9 @@ static std::shared_ptr<Aws::S3::S3Client> new_s3client(const S3URI& uri) {
     } else if (!config::object_storage_endpoint.empty()) {
         config.endpointOverride = config::object_storage_endpoint;
     }
+    if (!config::object_storage_region.empty()) {
```
Comment (caneGuy) [inline be/src/env/env_s3.cpp:207]: Does setting the AWS_REGION or AWS_DEFAULT_REGION environment variable work?

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by DorianZheng** [comment]
@stephen-shelby Plz update the doc: https://github.com/StarRocks/docs.zh-cn/pull/293/files

File: be/src/common/config.h [hunk 2]
```diff
@@ -662,6 +662,8 @@ CONF_Bool(rewrite_partial_segment, "true");
 CONF_String(object_storage_access_key_id, "");
 CONF_String(object_storage_secret_access_key, "");
 CONF_String(object_storage_endpoint, "");
+// Tencent cos needs to add region information
```
Comment (caneGuy) [inline be/src/common/config.h:665]: For oss we can also set this manually. how about changing the annotation to cover more cases?
and for aws ec2, if we set this then the AWS_REGION or region get by ec2 meta client can not take effect

File: be/src/env/env_s3.cpp (hunk 1 above)
Comment (caneGuy) [reply be/src/env/env_s3.cpp:207]: this is LGTM to use a be config

**Review by dirtysalt — APPROVED** [review]

File: be/src/common/config.h (hunk 2 above)
Comment (stephen-shelby) [reply be/src/common/config.h:665]: If anyone update the AWS_REGION env, just doen't set the region in be.conf? what do you think about it.

Comment (caneGuy) [reply be/src/common/config.h:665]: LGTM @stephen-shelby what i think about is that we need to mention this risk in the configuration instructions.

Comment (stephen-shelby) [reply be/src/common/config.h:665]: ok, I will explain it in the documentation about this config.

**Review by caneGuy — APPROVED** [review]

**Pull request merged**